### PR TITLE
Enable Domain Deletions

### PR DESCRIPTION
**Pull request by sunlnu**
# stream-registry PR

_&lt;Domain deletion was not supported earlier. After this change now we can delete the domains in stream registry&gt;_

### Added
* _&lt;Domain deletion enabled&gt;_
* _&lt;Domain deletion is added&gt;_

### Changed
* _&lt;Domain deletion enabled&gt;_
* _&lt;Domain deletion enabled&gt;
* _&lt;in particular, describe BACKWARD INCOMPATIBLE changes&gt;

### Deleted
* _&lt;Domain deletion disabled&gt;_

# PR Checklist Forms

- [ ] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java [hunk 1]
```diff
@@ -89,7 +89,8 @@ public List<Domain> findAll(Predicate<Domain> filter) {
 
   @PreAuthorize("hasPermission(#domain, 'DELETE')")
   public void delete(Domain domain) {
-    throw new UnsupportedOperationException("Domain deletion not currently supported.");
+    handlerService.handleDelete(domain);
+    domainRepository.delete(domain);
```
Comment (jamespfaulkner) [inline core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java:93]: Don't we need to worry ensure we maintain referential integrity here? Could this leave a lot of entities (`Stream`, `ProcessInputStream` etc) with references to domains that no longer exist?

Comment (sunlnu) [reply core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java:134]: correct.
Added

**Review by andykruth-ii — APPROVED** [review]

File: core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java [hunk 2]
```diff
@@ -89,7 +104,37 @@ public List<Domain> findAll(Predicate<Domain> filter) {
...
+    consumerView
+      .findAll(c -> c.getKey().getStreamKey().getDomainKey().equals(domain.getKey()))
+      .findAny()
+      .ifPresent(ex -> { throw errorHandler(); });
+
+    domainRepository.delete(domain);
   }
 
+  private static IllegalStateException errorHandler() {
+    return new IllegalStateException("Domain in use.");
```
Comment (jamespfaulkner) [inline core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java:138]: I'd add to the exception where the Domain is in use (`schema`, `process` etc)

Comment (sunlnu) [reply core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java:138]: ok , i was thinking the same . But not implemented , i can't remember, what was in my mind. 
Adding it .

**Review by jamespfaulkner — APPROVED** [review]

**Pull request merged**